pjsip/src/pjsip/sip_transport_tcp.c

```diff
@@ -1507,6 +1550,20 @@ static pj_bool_t on_connect_complete(pj_activesock_t *asock,
     return PJ_TRUE;
 }
 
+/* Transport connection timeout timer callback */
```

Comment:
How does the code protect against the following race condition

-   Timer pops just as the connection is about to fail
-    tcp_connect_timer starts, but is then interrupted by on_connect_complete, which destroys the tcp_stransport structure.
-    tcp_connect_timer continues, holding a pointer to potentially invalid/reused memory.
(if this is a problem, then I guess it affects the ka_timer as well, as I don't see any explicit locking+cb validation there either)